sphinxcontrib/confluencebuilder/builder.py

```diff
@@ -151,7 +152,18 @@ def get_relative_uri(self, from_, to, typ=None):
         return relative_uri(self.get_target_uri(from_),
                             self.get_target_uri(to, typ))
 
+    def register_parents(self, filename):
+        tree = self.env.get_doctree(filename)
+        for toctreenode in tree.traverse(addnodes.toctree):
+            for includefile in toctreenode['includefiles']:
```

Comment:
Sphinx core makes that assumption: https://github.com/sphinx-doc/sphinx/blob/master/sphinx/util/nodes.py

Check `inline_all_toctrees`